src/main/java/uk/gov/hmcts/sscs/service/ccd/ReadCoreCaseDataService.java

```diff
@@ -50,15 +52,35 @@ private CaseDetails get(EventRequestData eventRequestData, String serviceAuthori
         );
     }
 
-    private CaseData getCaseData(Object object) {
-
-        ObjectMapper mapper = new ObjectMapper();
-        mapper.enable(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY);
+    public CaseDetails getCcdCaseDetailsByAppealNumber(String appealNumber) {
+        log.info("Get CcdCaseDetails by appealNumber...");
+        EventRequestData eventRequestData = coreCaseDataService.getEventRequestData("appealCreated");
+        String serviceAuthorization = coreCaseDataService.generateServiceAuthorization();
 
-        try {
-            return mapper.convertValue(object, CaseData.class);
-        } catch (Exception e) {
-            throw new ApplicationErrorException("Error occurred when CaseDetails are mapped into CcdCase", e);
+        List<CaseDetails> caseDetailsList = getByAppealNumber(eventRequestData, serviceAuthorization, appealNumber);
+        CaseDetails caseDetails = null;
+        if (!caseDetailsList.isEmpty()) {
+            caseDetails = caseDetailsList.get(0);
         }
+        return caseDetails;
+    }
+
+    public CaseData getCcdCaseDataByAppealNumber(String appealNumber) {
+        log.info("Get CcdCaseData by appealNumber...");
+        CaseDetails caseDetails = getCcdCaseDetailsByAppealNumber(appealNumber);
+        return CcdUtil.getCaseData(caseDetails != null ? caseDetails.getData() : null);
+    }
+
+    private List<CaseDetails> getByAppealNumber(EventRequestData eventRequestData, String serviceAuthorization,
+                                                String appealNumber) {
+        log.info("Get getByAppealNumber...");
+        return coreCaseDataService.getCoreCaseDataApi().searchForCaseworker(
+                eventRequestData.getUserToken(),
+                serviceAuthorization,
+                eventRequestData.getUserId(),
+                eventRequestData.getJurisdictionId(),
+                eventRequestData.getCaseTypeId(),
+                ImmutableMap.of("case.subscriptions.appellantSubscription.tya", appealNumber)
+        );
```

Comment:
Log info doesn't contain any useful information. Do we need so many info logs?. Its better to change it to debug and include additional information.

Reply:
Removed the unnecessary info logs.